sematic/plugins/metrics_storage/sql/sql_metrics_storage.py

```diff
@@ -0,0 +1,263 @@
+            session.add_all(new_metric_labels)
+            session.add_all(metric_values)
+            session.commit()
+
+    def get_metrics(self, labels: MetricsLabels) -> Tuple[str, ...]:
+        with db().get_session() as session:
+            metric_names = (
+                session.query(MetricLabel.metric_name)
+                .distinct(MetricLabel.metric_name)
+                .filter(*_make_predicates_from_labels(labels))
+                .all()
+            )
+
+        return tuple(row[0] for row in metric_names)
+
+    def get_aggregated_metrics(
+        self,
+        filter: MetricsFilter,
+        group_by: Sequence[GroupBy],
+        rollup: Union[int, Literal["auto"], None],
+    ) -> MetricSeries:
+        # Early check as later queries fail when there are no rows.
+        try:
+            with db().get_session() as session:
+                session.query(MetricLabel.metric_id).filter(
+                    MetricLabel.metric_name == filter.name
+                ).limit(1).one()
+        except NoResultFound:
+            raise NoMetricError(filter.name, self.get_path())
+
+        predicates = [
+            MetricLabel.metric_id == MetricValue.metric_id,
+            MetricLabel.metric_name == filter.name,
+            MetricValue.metric_time >= filter.from_time,
+            MetricValue.metric_time < filter.to_time,
+        ]
+
+        predicates += _make_predicates_from_labels(filter.labels)
+
+        select_fields = [
+            MetricLabel.metric_name,
+            MetricLabel.metric_type,
+            func.sum(MetricValue.value),
+            func.count(MetricValue.value),
+        ]
+
+        n_basic_fields = len(select_fields)
+
+        extra_field_names = []
+
+        group_by_clauses = [
+            MetricLabel.metric_name,
+            MetricLabel.metric_type,
+        ]
+
+        time_range = filter.to_time - filter.from_time
+        # Ceil to make sure we always have fewer than _MAX_SERIES_POINTS
+        interval_seconds = math.ceil(time_range.total_seconds() / _MAX_SERIES_POINTS)
+
+        if isinstance(rollup, int):
+            interval_seconds = max(interval_seconds, rollup)
+            field_ = (
+                func.extract("epoch", MetricValue.metric_time)
+                / interval_seconds
+                * interval_seconds
+            )
+            select_fields.append(field_)
+            group_by_clauses.append(field_)
+            extra_field_names.append("timestamp")
+
+        for gb in group_by:
+            extra_field_names.append(gb.value)
+            field_ = MetricLabel.metric_labels[gb.value].astext
+
+            select_fields.append(field_)
+            group_by_clauses.append(field_)
+
+        with db().get_session() as session:
+            query = session.query(*select_fields).filter(*predicates)
+
+            if len(group_by_clauses) > 0:
+                query = query.group_by(*group_by_clauses)
+
+            if rollup == "auto":
```

Comment:
Please extract this if to an elif branch under `if isinstance(rollup, int):`.

Reply:
No because this needs to happen within the DB session.